### PR TITLE
fix 876

### PR DESCRIPTION
**Pull request by wind57**

### CODE REVIEW

File: spring-cloud-kubernetes-fabric8-config/src/test/java/org/springframework/cloud/kubernetes/fabric8/config/Fabric8SecretsPropertySourceTest.java [hunk 1]
```diff
@@ -73,9 +75,9 @@ public static void setUpBeforeClass() {
 
 	@Test
 	public void toStringShouldNotExposeSecretValues() {
-		String actual = this.propertySourceLocator.locate(this.environment).toString();
-
-		assertThat(actual).doesNotContain(SECRET_VALUE);
+		PropertySource<?> propertySource = this.propertySourceLocator.locate(this.environment);
+		assertThat(propertySource.toString()).doesNotContain(SECRET_VALUE);
+		assertThat(propertySource.getProperty("secretName")).isEqualTo("secretValue");
```
Comment (wind57) [inline spring-cloud-kubernetes-fabric8-config/src/test/java/org/springframework/cloud/kubernetes/fabric8/config/Fabric8SecretsPropertySourceTest.java:80]: adds one more assert in here, tests remains the same

File: spring-cloud-kubernetes-fabric8-config/src/test/java/org/springframework/cloud/kubernetes/fabric8/config/Fabric8BootstrapConfigurationTests.java [hunk 2]
```diff
@@ -129,4 +130,39 @@ public void secretsOnlyPresent() {
 
 	}
 
+	// tests that @ConditionalOnCloudPlatform(CloudPlatform.KUBERNETES) has the desired
```
Comment (wind57) [inline spring-cloud-kubernetes-fabric8-config/src/test/java/org/springframework/cloud/kubernetes/fabric8/config/Fabric8BootstrapConfigurationTests.java:133]: these two below prove that `@ConditionalOnCloudPlatform(CloudPlatform.KUBERNETES` enable/disable brings or not, the needed beans

**Comment by wind57** [comment]
@ryanjbaxter this uses the boot api as discussed and it is ready to review. It will fix

- [this issue](https://github.com/spring-cloud/spring-cloud-kubernetes/issues/876)
- [and this one](https://github.com/spring-cloud/spring-cloud-kubernetes/issues/883)

**Comment by wind57** [comment]
@ryanjbaxter this one ready to be reviewed also (sorry for pinging you so much)... trying to clean-up some small things a bit, until I move to some changes and fixes in the secrets implementation. thank you!

**Comment by ryanjbaxter** [comment]
No worries, as always I appreciate it!

**Pull request merged**